docs/source/advanced_install.rst

```diff
@@ -121,28 +133,82 @@ Prompt for VS 2019.
     If you want to use ``pip`` instead of ``conda``, the following commands can
     be used::
 
-    $ pip install cmake scikit-build pybind11
-    $ pip install torch -f https://download.pytorch.org/whl/torch_stable.html
+        $ pip install cmake scikit-build pybind11
+        $ pip install torch -f https://download.pytorch.org/whl/torch_stable.html
+
 
 Installing and compiling
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 Once the dependencies are in place, the following command can be used for
-compiling and installing the Python package::
+compiling. Here we assume that you have already cloned the directory
+and changed into it::
+
+    $ git clone https://github.com/IBM/aihwkit.git
+    $ cd aihwkit
+
+You can typically install requirements by (but see above for more
+specific details)::
+
+    $ pip install -r requirements.txt
+    $ pip install -r requirements-dev.txt
+    $ pip install -r requirements-examples.txt
+
+
+Without GPU support (with OpenBLAS):
+  This uses the OpenBLAS library for fast numerical computations::
+
+    $ make build
+
+  .. note::
 
-    $ pip install -v aihwkit
+    Note that openblas needs to be installed, e.g. with::
+        $ conda install openblas
 
-This command will:
+Without GPU support (with MKL):
+  This uses the Intel MKL library instead of the OpenBlas library::
+
+    $ make build_mkl
+
+  .. note::
+     Note that MKL needs to be installed and environment variable
+     ``MKLROOT`` set if not in standard folders. E.g. with::
+
+         $ conda install -c intel mkl mkl-devel mkl-static mkl-include
+
+With GPU support:
+  The CUDA library needs to be set up properly so that the compiler
+  can find it (you may need to set ``CUDA_HOME``). Please refer to the
+  installation instructions. This also uses MKL
+  as default, whihc thus needs to be installed (see above). Then::
+
+      $ make build_cuda
+
+  If you know your CUDA architecture, then you can give it directly
+  (which will result typically in a much quicker initially loading time)::
+
+      $ make build_cuda flags='-DRPU_CUDA_ARCHITECTURES="60"'
 
-* download the source tarball for the library.
-* invoke ``scikit-build``
-* which in turn will invoke ``cmake`` for the compilation.
-* execute the commands in verbose mode, for helping troubleshooting issues.
-* install the Python package.
 
 If there are any issue with the dependencies or the compilation, the output
 of the command will help diagnosing the issue.
 
+In-place installation
+~~~~~~~~~~~~~~~~~~~~~
+
+If you want install the library inside the cloned directory (see also
+:doc:`developer_install`), it is more convineint for developers. For
+that simply replace the above make commands with ``build_inplace``,
+e.g.::
+
+    $ make build_inplace_cuda
+
+Here you need to make sure that the ``PYTHONPATH`` is set to the ``src``
+sub-directory of the cloned directory, e.g. by (when being in the cloned directory)::
```

Comment:
This sentence is confusing. Do you mean: e.g, when you are in the cloned directory.?

Reply:
changed to base dir